### PR TITLE
[FLINK-17339][table] Change default planner to blink

### PR DESCRIPTION
**Pull request by KurtYoung**
## What is the purpose of the change

Change the default planner to blink, and update test cases caused by this.

## Verifying this change

This change is already covered by existing tests

### CODE REVIEW

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/EnvironmentSettings.java [hunk 1]
```diff
@@ -159,8 +159,8 @@ public boolean isStreamingMode() {
 		private static final String BLINK_PLANNER_FACTORY = "org.apache.flink.table.planner.delegation.BlinkPlannerFactory";
 		private static final String BLINK_EXECUTOR_FACTORY = "org.apache.flink.table.planner.delegation.BlinkExecutorFactory";
 
-		private String plannerClass = OLD_PLANNER_FACTORY;
```
Comment (libenchao) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/EnvironmentSettings.java:162]: also update java doc for `EnvironmentSettings`?

**Comment by KurtYoung** [comment]
@libenchao Thanks for the reviewing, I've addressed your comments.

**Comment by libenchao** [comment]
@KurtYoung Thanks for addressing my comments. It seems that you didn't notice 

> `TpcdsTestProgram` and `BatchSQLTestProgram` is missing 

in the comment.

**Comment by KurtYoung** [comment]
@libenchao I've noticed them. I think it's ok to use blink planner explicitly

**Comment by libenchao** [comment]
@KurtYoung Thanks for the explain, the changes LGTM now.
BTW, we should also update the documentation about this change. (I think you must be aware of this, and planned to do it in another issue. Just a reminder)

**Comment by KurtYoung** [comment]
Yes, it's FLINK-17340

File: flink-examples/flink-examples-table/src/main/scala/org/apache/flink/table/examples/scala/StreamSQLExample.scala [hunk 2]
```diff
@@ -49,12 +49,16 @@ object StreamSQLExample {
     val env = StreamExecutionEnvironment.getExecutionEnvironment
     val tEnv = if (planner == "blink") {  // use blink planner in streaming mode
```
Comment (JingsongLi) [inline flink-examples/flink-examples-table/src/main/scala/org/apache/flink/table/examples/scala/StreamSQLExample.scala:50]: Minor: `EnvironmentSettings.Builder settingsBuilder = EnvironmentSettings.newInstance().inStreamingMode();` to reuse `StreamTableEnvironment.create` code.
`StreamSQLExample.java` too.

File: flink-connectors/flink-connector-cassandra/pom.xml [hunk 3]
```diff
@@ -172,7 +172,7 @@ under the License.
 		<!-- A planner dependency won't be necessary once FLIP-32 has been completed. -->
 		<dependency>
 			<groupId>org.apache.flink</groupId>
-			<artifactId>flink-table-planner_${scala.binary.version}</artifactId>
+			<artifactId>flink-table-planner-blink_${scala.binary.version}</artifactId>
```
Comment (JingsongLi) [inline flink-connectors/flink-connector-cassandra/pom.xml:175]: Why just change cassandra?
Maybe scope to test and remove above comments?

File: flink-python/pyflink/table/tests/test_table_environment_api.py [hunk 4]
```diff
@@ -275,16 +275,16 @@ def test_create_table_environment_with_blink_planner(self):
 
         self.assertEqual(
             planner.getClass().getName(),
-            "org.apache.flink.table.planner.StreamPlanner")
+            "org.apache.flink.table.planner.delegation.StreamPlanner")
 
         t_env = StreamTableEnvironment.create(
-            environment_settings=EnvironmentSettings.new_instance().use_blink_planner().build())
+            environment_settings=EnvironmentSettings.new_instance().use_old_planner().build())
```
Comment (JingsongLi) [inline flink-python/pyflink/table/tests/test_table_environment_api.py:281]: Why change to old? test name is `test_create_table_environment_with_blink_planner`? maybe should change method name?

File: flink-examples/flink-examples-table/src/main/scala/org/apache/flink/table/examples/scala/StreamSQLExample.scala (hunk 2 above)
Comment (KurtYoung) [reply flink-examples/flink-examples-table/src/main/scala/org/apache/flink/table/examples/scala/StreamSQLExample.scala:50]: I think current is also fine.

File: flink-connectors/flink-connector-cassandra/pom.xml (hunk 3 above)
Comment (KurtYoung) [reply flink-connectors/flink-connector-cassandra/pom.xml:175]: I'm not sure about this, did you see any problem with it?

File: flink-python/pyflink/table/tests/test_table_environment_api.py (hunk 4 above)
Comment (KurtYoung) [reply flink-python/pyflink/table/tests/test_table_environment_api.py:281]: The test is trying to cover all 3 situations, 1) default behavior 2) use blink planner explicitly 3) use old planner explicitly.

File: flink-connectors/flink-connector-cassandra/pom.xml (hunk 3 above)
Comment (JingsongLi) [reply flink-connectors/flink-connector-cassandra/pom.xml:175]: It is not a big deal.

**Pull request closed without merging**